Realm/Realm/DatabaseTypes/RealmCollectionBase.cs

```diff
@@ -39,17 +41,20 @@ public abstract class RealmCollectionBase<T>
           IThreadConfined,
           IMetadataObject
     {
+        private readonly List<NotificationCallbackDelegate<T>> _callbacks = new List<NotificationCallbackDelegate<T>>();
```

Comment:
Unfortunately, this change also reorders the members of `RealmCollectionBase` because they were all over the place. Apart from the reordering, the only other change is the addition of the attributes.